src/actions/authorization.js

```diff
@@ -0,0 +1,42 @@
+import superagent from "superagent";
+import { url as baseUrl } from "../url";
+
+export const LOGIN_OR_SIGNUP_ERROR = "LOGIN_OR_SIGNUP_ERROR";
+export const LOGOUT = "LOGOUT";
+
+//this action is created on server side
+export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
+
+const loginError = error => {
+  return {
+    type: LOGIN_OR_SIGNUP_ERROR,
+    payload: error
+  };
+};
+export const logOut = () => {
+  return {
+    type: LOGOUT
+  };
+};
+
+export const loginSignupFunction = (
+  type,
+  name,
+  password,
+  history
+) => async dispatch => {
+  const url = `${baseUrl}/${type}`;
+  try {
+    const response = await superagent.post(url).send({ name, password });
```

Comment:
Yep, it is a convenient way to use libs for requests. But I think knowing of basic browser API better than that.
Try to use `fetch` it is enough modern API for requests (just see how works XmlHttpRequest it was a pile of pain), now it's good for using just browser methods. It is supported by all modern browsers if you need support of older just add `whatwg-fetch` polyfill.